### DISCUSSION TITLE
REST - Wrong request method for updates

### DISCUSSION THREAD

**Discussion by wojciechkrol** [payloadcms/payload#651] [Feature Requests & Ideas]
Hello devs,

I see you are using PUT request method to update collection data. PUT method should be used to create or update an existing item and requires the entire resource to be sent (update should replace the whole document).

In my opinion, you should change this method to PATCH which covers your usage. Why?

- Right now using PUT request when a resource does not exist, it does not create a new one when it should.
- We can send partial data (update just one field) and it updates only the selected field instead of replacing the whole document.
- See: https://stackoverflow.com/questions/28459418/use-of-put-vs-patch-methods-in-rest-api-real-life-scenarios

Best regards,

**Comment by jmikrut** [comment]
Hey @wojciechkrol — 

You're totally right. This would be pretty significant breaking change though. I think if we were to make this change, it'd be before 1.0.

We could potentially open an endpoint for `PATCH` and leave the `PUT` alone for a while, with a corresponding log warning every time a `PUT` endpoint is used.

Then once we hit `1.0.0`, we could retire the old `PUT` endpoint. 

What do you think?

**Comment by wojciechkrol** [comment]
Hey @jmikrut,

I totally agree you shouldn't make breaking changes to keep existing projects working without refactoring. I think you can open an endpoint for `PATCH` requests and let people migrate as soon as possible.

Do you have any roadmap for `1.0.0`?

**Reply by jmikrut** [reply]
Sure do—we just posted an update about this 😈 

https://payloadcms.com/blog/upcoming-1-0

**Comment by wojciechkrol** [comment]
Hey @jmikrut,

I see you hit `1.0.4` 🎉. Did you make an update to allow PATCH requests?

**Comment by wojciechkrol** [comment]
Hello @jmikrut 

Could you enable PATCH request as an alternative for PUT?

**Reply by jmikrut** [reply]
Hey @wojciechkrol — do you mean just simply open another endpoint for `PATCH`, which would be duplicative to `PUT`? At this point we have decided to avoid breaking changes to our existing `PUT` endpoint, but exposing an additional `PATCH` could be doable if the question is just semantics.

What do you think?

**Reply by wojciechkrol** [reply]
Hey @jmikrut - Yes, just simply open another endpoint for `PATCH` which will duplicate `PUT` for now. Your previous idea to open `PATCH` and mark `PUT` as deprecated was great. You can implement `PUT` properly later on or just remove it.

For now `PATCH` fits better than `PUT` for current functionality.

**Reply by jmikrut** [reply]
OK - this is good stuff.

We would happily accept a PR for this if you're up for it! Would be simple. Need to add here:

https://github.com/payloadcms/payload/blob/master/src/collections/buildEndpoints.ts

Wanna take a shot?
